indexer/services/comlink/src/controllers/api/v4/perpetual-positions-controller.ts

```diff
@@ -150,6 +156,143 @@ class PerpetualPositionsController extends Controller {
       }),
     };
   }
+
+  @Get('/parentSubaccountNumber')
+  async listPositionsForParentSubaccount(
+    @Query() address: string,
+      @Query() parentSubaccountNumber: number,
+      @Query() status?: PerpetualPositionStatus[],
+      @Query() limit?: number,
+      @Query() createdBeforeOrAtHeight?: number,
+      @Query() createdBeforeOrAt?: IsoString,
+  ): Promise<PerpetualPositionResponse> {
+    // Get subaccountIds for all child subaccounts of the parent subaccount
+    // Create a record of subaccountId to subaccount number
+    const childIdtoSubaccountNumber: Record<string, number> = {};
+    getChildSubaccountNums(parentSubaccountNumber).forEach(
+      (subaccountNum: number) => {
+        childIdtoSubaccountNumber[SubaccountTable.uuid(address, subaccountNum)] = subaccountNum;
+      },
+    );
+    const childSubaccountIds: string[] = Object.keys(childIdtoSubaccountNumber);
```

Comment:
nit: I think this should be in a separate function since we need it for every one of these parent subaccount number endpoints

Reply:
There does exist a helper function to getChildSubaccountIds already but I also need to create the map above for this controller specifically, so this line here does not make a difference